classes/PaymentModule.php

```diff
@@ -639,6 +638,10 @@ public function validateOrder($id_cart, $id_order_state, $amount_paid, $payment_
 
                             $values['tax_incl'] = $order->total_products_wt - $total_reduction_value_ti;
                             $values['tax_excl'] = $order->total_products - $total_reduction_value_tex;
+                            if (1 == $voucher->free_shipping) {
+                                 $values['tax_incl'] = $order->total_products_wt + $order->total_shipping_tax_incl - $total_reduction_value_ti;
+                                 $values['tax_excl'] = $order->total_products + $order->total_shipping_tax_excl - $total_reduction_value_tex;  
+                            }
```

Comment:
Please look carefully at the code block in my previous comment. In the if statement, I added only the specific operations. The common part of the calculation is outside (before the if statement)